src/components/common/Home/CreateTargetForm/index.js

```diff
@@ -0,0 +1,170 @@
+import React, { useEffect } from 'react';
+import { shallowEqual, useSelector } from 'react-redux';
+import { Link, useHistory } from 'react-router-dom';
+
+import { REJECTED, PENDING } from 'constants/actionStatusConstants';
+
+import Loading from 'components/common/Loading';
+import Input from 'components/common/Input';
+import SelectInput from 'components/common/SelectInput';
+import Target from 'assets/target.png';
+import { createTarget, getTargetTopics } from 'state/actions/targetActions';
+import { createTarget as createTargetValidations } from 'utils/constraints';
+import { FormattedMessage } from 'react-intl';
+import routes from 'constants/routesPaths';
+import {
+  useValidation,
+  useStatus,
+  useTextInputProps,
+  useForm,
+  useDispatch,
+  useNewTarget
+} from '../../../../hooks';
+
+const fields = {
+  title: 'title',
+  lat: 'lat',
+  lng: 'lng',
+  radius: 'radius',
+  topic_id: 'topic_id'
+};
+
+const CreateTargetForm = () => {
+  const validator = useValidation(createTargetValidations);
+  const { status, error } = useStatus(createTarget);
+  const createTargetRequest = useDispatch(createTarget);
+  const getTopicsRequest = useDispatch(getTargetTopics);
+  const topics = useSelector(state => state.targetReducer.topics, shallowEqual);
+  const history = useHistory();
+  const newTarget = useNewTarget();
+
+  useEffect(() => {
+    getTopicsRequest();
+  }, [getTopicsRequest]);
+
+  const initialValues = {
+    title: '',
+    lat: '',
+    lng: '',
+    topic_id: '',
+    radius: ''
+  };
+
+  const {
+    values,
+    errors,
+    submitted,
+    handleValueChange,
+    handleSubmit,
+    handleFocus,
+    handleBlur,
+    activeFields,
+    touched
+  } = useForm(
+    {
+      initialValues,
+      onSubmit: createTargetRequest,
+      validator,
+      validateOnBlur: true
+    },
+    [createTargetRequest]
+  );
+
+  const inputProps = useTextInputProps(
+    handleValueChange,
+    handleFocus,
+    handleBlur,
+    values,
+    errors,
+    activeFields,
+    touched
+  );
+
+  useEffect(() => {
+    if (
```

Comment:
Would be good leaving a comment above these conditions to explain what they're doing, since they would look pretty confusing for someone new 😅